### PR TITLE
feat: Update several packages in Meshery

### LINKED ISSUE #9136: Upgrade go mod packages
Meshery is dire need to update its Go packages, but there were a lot of packages that were causing issues directly and indirectly.

We rely mostly on the Kubernetes API packages, so there were frequent changes done in the last year going from Kubernetes API `v0.25.0` to `v0.27.0`.

The problem now is making sure that everything syncs and matches across the multitude of repos that we currently use these APIs in.

- [ ] `v1.StatusApplyConfiguration` is deprecated
- [ ] Look into updating `taskq` from `v3` to `v4` and update the relevant files

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by nebula-aac**
**Notes for Reviewers**

This PR fixes #9136 

Opening up a PR to track the dependencies issues.

This branch also picks up the latest commits from our forked kompose open PR: https://github.com/meshery/kompose/pull/3

Blocking:

I was testing `taskq` v4, but may revert to `v3` if there is no problems.

```bash
server/handlers/handler_instance.go:66:30: not enough arguments in call to taskq.RegisterTask
        have (*taskq.QueueConfig)
        want (string, *taskq.TaskConfig)
server/handlers/handler_instance.go:68:12: cannot use h.CollectStaticMetrics (value of type func(config *"github.com/layer5io/meshery/server/models".SubmitMetricsConfig) error) as taskq.Handler value in struct literal: func(config *"github.com/layer5io/meshery/server/models".SubmitMetricsConfig) error does not implement taskq.Handler (missing method HandleJob)
server/handlers/load_test_handler.go:578:14: h.task.WithArgs undefined (type *taskq.Task has no field or method WithArgs)
```

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
I made a futile attempt at reconciling the merge conflict.

**Comment by leecalcote** [comment]
@nebula-aac, please do signal when you consider this one is ready to go. There’s lots of good cleanup and important maintenance in here.

File: install/docker/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21 as meshery-server
+FROM golang:1.20 as meshery-server
```
Comment (MUzairS15) [inline install/docker/Dockerfile:1]: Intentional?

**Comment by MUzairS15** [comment]
I updated `k8s.io/client-go and k8s.io/kubectl ` to `v0.28.3, `github.com/layer5io/meshkit` to `v0.6.75` and `github.com/layer5io/meshsync` to `v0.6.22`.
The server now builds fine, but need to perform some additional checks.

// @nebula-aac @leecalcote

**Comment by leecalcote** [comment]
> I updated `k8s.io/client-go and k8s.io/kubectl ` to `v0.28.3, `github.com/layer5io/meshkit`to`v0.6.75`and`github.com/layer5io/meshsync`to`v0.6.22`. The server now builds fine, but need to perform some additional checks.
> 
> // @nebula-aac @leecalcote

@MUzairS15 thank you. Perhaps, while you can pull this branch into your other current dev branch so that you are implicitly testing while enhancing Connections and other areas...

**Comment by leecalcote** [comment]
resyncing

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**